src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2624,7 +2624,7 @@ private void EmitDefaultValue(TypeSymbol type, bool used, SyntaxNode syntaxNode)
             if (used)
             {
                 // default type parameter values must be emitted as 'initobj' regardless of constraints
-                if (!type.IsTypeParameter())
+                if (!type.IsTypeParameter() && type.SpecialType != SpecialType.System_Decimal)
```

Comment:
Another type like Decimal is "DateTime". Please add a similar test for that. 
C# does not allow DateTime constants so GetDefaultValue will not produce a constant for it (in VB it would), and that saves us here.
However - just in case if anything changes we need a similar test with DateTime.